### PR TITLE
BCI mood reading, improves mood event code

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

BCIs now send signals for new moodlets!

- [ ] sanitize out the spans either by making moodlets hold a raw version of their text or a helper for stripping it (probably the former)

## Why It's Good For The Game

BCI triggering from specific moods sounds really funny. You could hook stuff up to say, your nicotine addiction pretty easily with this

## Changelog
:cl:
expansion: BCIs now read incoming moodlets!
/:cl:

### CODE REVIEW

**Comment by Mothblocks** [comment]
Hook removing moods as well, no?

File: code/modules/wiremod/shell/brain_computer_interface.dm [hunk 1]
```diff
@@ -260,6 +264,19 @@
 	parent.cell.give(shock_damage * 2)
 	to_chat(source, span_notice("You absorb some of the shock into your [parent.name]!"))
 
+/obj/item/circuit_component/bci_core/proc/on_mood_event(datum/source, category, type, ...)
+	if(source != user?.resolve())
```
Comment (Mothblocks) [inline code/modules/wiremod/shell/brain_computer_interface.dm:268]:
```suggestion
	if(IS_WEAKREF_OF(source, user))
```

File: code/modules/wiremod/shell/brain_computer_interface.dm [hunk 2]
```diff
@@ -260,6 +264,19 @@
 	parent.cell.give(shock_damage * 2)
 	to_chat(source, span_notice("You absorb some of the shock into your [parent.name]!"))
 
+/obj/item/circuit_component/bci_core/proc/on_mood_event(datum/source, category, type, ...)
```
Comment (Mothblocks) [inline code/modules/wiremod/shell/brain_computer_interface.dm:267]: SIGNAL_HANDLER

File: code/modules/wiremod/shell/brain_computer_interface.dm [hunk 3]
```diff
@@ -260,6 +264,19 @@
...
+	if(source != user?.resolve())
+		CRASH("A BCI's user was changed and it didn't notice.")
+	var/datum/mood_event/temp_event
+	var/list/params = args.Copy(4)
+	params.Insert(1, parent)
+	//builds the moodlet text
+	temp_event = new type(arglist(params))
```
Comment (Mothblocks) [inline code/modules/wiremod/shell/brain_computer_interface.dm:274]: *wipes sweat*

**Comment by tralezab** [comment]
Sure I can do that, all the signal stuff is already around

File: code/modules/wiremod/shell/brain_computer_interface.dm [hunk 4]
```diff
@@ -165,6 +165,9 @@
 
 	var/datum/port/output/user_port
 
+	var/datum/port/output/mood
+	var/datum/port/output/mood_trigger
```
Comment (Mothblocks) [inline code/modules/wiremod/shell/brain_computer_interface.dm:169]: Doesn't look like you create these anywhere?

**Review by Mothblocks — CHANGES_REQUESTED** [review]
I'd rather this be a BCI-specific component, but I think that PR isn't merged yet, so eh.

**Comment by Mothblocks** [comment]
Actually, let's wait for that PR to be merged.

**Comment by Time-Green** [comment]
Wait could this be used to detect antag moodlets, like cult conversions? Would be pretty cool

**Comment by Thunder12345** [comment]
Antag test, antag test

**Comment by Mothblocks** [comment]
that sounds funny let's let it play out first before blacklisting moodlets

**Pull request closed without merging**